src/components/FilterPanel.jsx

```diff
@@ -15,41 +15,61 @@ import AmountFilter from './AmountFilter';
 import GradeLevelFilter from './GradeLevelFilter';
 import MajorFilter from './MajorFilter';
 import CloseIcon from '@mui/icons-material/Close';
+import WarningAmberOutlinedIcon from '@mui/icons-material/WarningAmberOutlined';
 import PropTypes from 'prop-types';
 
 export default function FilterPanel({ onClose }) {
-  const [{ minAmount, maxAmount, grades, majors }, setQueryParam] =
-    useQueryParams();
+  const [params, setQueryParams] = useQueryParams();
+
+  const [minAmount, setMinAmount] = useState(params.minAmount);
+  const [maxAmount, setMaxAmount] = useState(params.maxAmount);
+  const [grades, setGrades] = useState(params.grades);
+  const [majors, setMajors] = useState(params.majors);
 
   const filters = {
-    Major: (
-      <MajorFilter
-        majors={majors}
-        onSelect={(m) => setQueryParam('majors', m)}
-        onDelete={(m) =>
-          setQueryParam(
-            'majors',
-            majors.filter((major) => major !== m)
-          )
-        }
-      />
-    ),
-    Amount: (
-      <AmountFilter
-        min={minAmount ?? 0}
-        max={maxAmount ?? 0}
-        onMinChange={(val) => setQueryParam('minAmount', val)}
-        onMaxChange={(val) => setQueryParam('maxAmount', val)}
-      />
-    ),
-    'Grade Level': (
-      <GradeLevelFilter
-        grades={new Set(grades)}
-        changeFn={(e) => setQueryParam('grades', e)}
-      />
-    ),
+    Major: {
+      comp: (
+        <MajorFilter
+          majors={majors}
+          onSelect={(m) => setMajors(m)}
```

Comment:
```suggestion
          onSelect={setMajors}
```

Reply:
I'm confused by how this works. If a major is selected should it be **added** to the `majors` list?  I.e. something like
```suggestion
          onSelect={(m) => {if (!majors.includes(m)) setMajors([...majors, m]);}}
```

Otherwise if the `MajorFilter` component takes care of updating this list why is this called `onSelect` and not something like `onChange`? And why do we have a separate `onDelete` function if `MajorFilter` can just call that change function? The behavior and naming of these functions is confusing and inconsistent.

Reply:
No need to check if majors includes the major. Not sure how, but it already does that on its own.